onnxruntime/core/session/inference_session.cc

```diff
@@ -415,96 +415,72 @@ class InferenceSession::Impl {
                   "Unexpected input data type. Actual: (" + actual_name + ") , expected: (" + expected_name + ")");
   }
 
-  common::Status ValidateInputTypes(const std::vector<std::string>& feed_names,
-                                    const std::vector<MLValue>& feeds) {
+  common::Status ValidateInputs(const std::vector<std::string>& feed_names,
+                                const std::vector<MLValue>& feeds) {
     const auto begin_names = feed_names.cbegin();
     const auto end_names = feed_names.cend();
-    for (auto& arg : input_def_list_) {
+    std::unordered_set<ptrdiff_t> required_feed_ids;
```

Comment:
You could also use a vector of Booleans here (of same size as feeds) for direct (indexed) access instead of hash-lookup.